pydantic/utils.py

```diff
@@ -285,6 +290,18 @@ def is_callable_type(type_: AnyType) -> bool:
     return type_ is Callable or getattr(type_, '__origin__', None) is Callable
 
 
+if sys.version_info >= (3, 7):
+
+    def is_literal_type(type_: AnyType) -> bool:
+        return Literal is not None and getattr(type_, '__origin__', None) is Literal
+
+
+else:
+
+    def is_literal_type(type_: AnyType) -> bool:
```

Comment:
In python 3.6, Literal is a little more awkward to check for. This cast-to-Literal approach is the only version that I could produce that I think would work in general. Let me know if you would rather structure the code differently than doing a version check.